code/controllers/subsystems/atoms.dm

```diff
@@ -96,6 +100,17 @@ SUBSYSTEM_DEF(atoms)
 
 	return qdeleted || QDELING(A)
 
+/datum/controller/subsystem/atoms/proc/GetArguments(atom/A, list/mapload_arg, created=TRUE)
+	var/list/arguments = mapload_arg.Copy()
+	if(created && created_atoms[A])
+		arguments += created_atoms[A]
+	if(override_populate_parts && istype(A, /obj/machinery))
```

Comment:
Should probably use a proc-override to acquire additional arguments rather than `istype()`-checks.
Open-closed principle and all that.

Reply:
Thing it's worth removing override_populate_parts entirely and leaving it up to mod implementor to override GetArguments()?

Reply:
It does look like the vars wouldn't be needed with a carefully implemented `/atom/proc/AdjustInitArguments(var/list/arguments)` implementation (or similar)

On the other hand, having an `adjust_init_arguments `-var (or config-setting) would mean the proc-calls could be skipped on servers that don't use it at all.
I.e:
```
/atom/proc/AdjustInitializeArguments(list/arguments)
    // Lists are passed by reference so can simply modify the arguments list without returning it

/datum/controller/subsystem/atoms
    var/adjust_init_arguments = TRUE/FALSE

/datum/controller/subsystem/atoms/proc/GetArguments(atom/A, list/mapload_arg, created=TRUE)
	var/list/arguments = mapload_arg.Copy()
	if(created && created_atoms[A])
		arguments += created_atoms[A]
	if(adjust_init_arguments)
        A.AdjustInitializeArguments(arguments)
    return arguments
```